api/users/views.py

```diff
@@ -333,6 +334,28 @@ def get_queryset(self):
         )
 
 
+class UserOSFGroups(JSONAPIBaseView, generics.ListAPIView, UserMixin, ListFilterMixin):
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+    )
+    required_read_scopes = [CoreScopes.OSF_GROUPS_READ]
+    required_write_scopes = [CoreScopes.NULL]
+
+    model_class = apps.get_model('osf.OSFGroup')
+    serializer_class = OSFGroupSerializer
+    view_category = 'users'
+    view_name = 'user-osf-groups'
+    ordering = ('-modified', )
+
+    def get_default_queryset(self):
+        return self.get_user().osf_groups
```

Comment:
Since there are not really restrictions on who can view osf_groups, this just returns all OSF Groups that a user belongs to (member or manager, doesn't matter)